algorithms/src/msm/variable_base/mod.rs

```diff
@@ -0,0 +1,73 @@
+// Copyright (C) 2019-2021 Aleo Systems Inc.
+// This file is part of the snarkVM library.
+
+// The snarkVM library is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+
+// The snarkVM library is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+// GNU General Public License for more details.
+
+// You should have received a copy of the GNU General Public License
+// along with the snarkVM library. If not, see <https://www.gnu.org/licenses/>.
+
+use snarkvm_curves::traits::AffineCurve;
+use snarkvm_fields::{PrimeField, Zero};
+
+mod standard;
+
+pub struct VariableBaseMSM;
+
+impl VariableBaseMSM {
+    #[allow(unused)]
```

Comment:
Can you use a `deprecated` flag here so it is not used?

Reply:
It's a private(unexposed) method, and serves as a reference implementation for tests that is more likely to be correct due to simplicity. It is used in tests. I think `deprecated` flag is not appropriate, if that's fine with you.

Reply:
Was also renamed to `naive` rather than `basic`, which I think is more appropriate.

Reply:
Just realized it's private. Can you move it below so it's clearly private?